### PR TITLE
Add Toilets icons to search results and place page

### PR DESCRIPTION
**Pull request by dvdmrtnz**
Add Toilets icons to search results and place page. Also lower priority of toilets.

* Closes https://github.com/organicmaps/organicmaps/issues/7102
* Closes https://github.com/organicmaps/organicmaps/issues/2849
* Related to https://github.com/organicmaps/organicmaps/issues/1648
* Related to https://github.com/organicmaps/organicmaps/issues/2119

### CODE REVIEW

**Comment by Jean-BaptisteC** [comment]
Not sure is needed to add a section in place page for that icon at the top is enough

**Comment by pastk** [comment]
The `WC` icon will be recognisable in English speaking and Western European countries mostly, its not something established worldwide.

Is there an emoji with main and woman like our toilet POI icon? I think it'll be much better.

File: data/styles/clear/include/priorities_4_overlays.prio.txt [hunk 1]
```diff
@@ -1137,7 +1137,6 @@ shop-wholesale                                      # icon z17- (also has captio
 === 500
 
 amenity-shower                                      # icon z17- (also has caption(optional) z17-)
-amenity-toilets                                     # icon z18- (also has caption(optional) z18-)
```
Comment (pastk) [inline data/styles/clear/include/priorities_4_overlays.prio.txt:1140]: I think toilets should be still above bbq (imagine a park with bbq places and a toilets block), outdoor seating, masts, silos and storage tanks still.

So either move them down too or just raise parkings. I think no one thought that there could be a parking with a toilets tag :)

**Review by pastk — COMMENTED** [review]
I also think that there is no need for a separate PP line. An emoji is enough.

**Comment by pastk** [comment]
Also looks like a lot of small boilerplate changes required in many different files to add each "POI attribute" like atm, toilets, etc.

I wonder if it could be streamlined/simplified in the future, as we're going to add many more attributes I think..

**Comment by dvdmrtnz** [comment]
> The `WC` icon will be recognisable in English speaking and Western European countries mostly, its not something established worldwide.
> 
> Is there an emoji with main and woman like our toilet POI icon? I think it'll be much better.

Yes, there is: 🚻

**Comment by dvdmrtnz** [comment]
> > Yes, there is: 🚻
> 
> This particular emoji is hard to recognize IMO... Is there one w/o the blue box?

Not that I know of. That’s the one there is

**Comment by pastk** [comment]
> > Yes, there is: 🚻

Maybe it doesn't look good on my mdpi screen, but will be alright on higher dpi devices...

**Review by vng — APPROVED** [review]
LGTM.
@pastk So before this change (current master) toilet displaces parking? :)

**Comment by vng** [comment]
> I wonder if it could be streamlined/simplified in the future, as we're going to add many more attributes I think..

At least rewrite with arrays of tags - should simplify the code.

**Comment by dvdmrtnz** [comment]
I am going to change the approach here. Instead of lowering the priority of `amenity=toilets`, I am going to add a new type `toilets=yes` and remove them from `replaced_tags.txt`. This way we make sure that the subtag `toilets=yes` never overrides the main `amenity` tag.

**Comment by dvdmrtnz** [comment]
> Not sure is needed to add a section in place page for that icon at the top is enough

> I also think that there is no need for a separate PP line. An emoji is enough.

Should I remove the Toilets attribute line in the place page then?

Changed emoji to 🚻.

**Comment by pastk** [comment]
> 
> Should I remove the Toilets attribute line in the place page then?
> 

What was your original intention of adding this line given that it "duplicates" the emoji?
If you agree its a duplication then yes, remove it please.

**Comment by dvdmrtnz** [comment]
> > Should I remove the Toilets attribute line in the place page then?
> 
> What was your original intention of adding this line given that it "duplicates" the emoji? If you agree its a duplication then yes, remove it please.

For consistency, For example WiFi is also duplicated in description and attribute line:

**Comment by dvdmrtnz** [comment]
> I am going to change the approach here. Instead of lowering the priority of `amenity=toilets`, I am going to add a new type `toilets=yes` and remove them from `replaced_tags.txt`. This way we make sure that the subtag `toilets=yes` never overrides the main `amenity` tag.

I have added toilets-yes to categories.txt but the search for this type is not working. Does this need to be added somewhere else for search to work? Can I get some help here?

**Comment by pastk** [comment]
> > What was your original intention of adding this line given that it "duplicates" the emoji? If you agree its a duplication then yes, remove it please.
> 
> For consistency, For example WiFi is also duplicated in description and attribute line:

Yeap, a good point.
I think it was intended to display various "internet options", like wired, terminal, etc. But its stuck with a "WiFi" option only.

```
  /// TODO: Support all possible Internet types in UI. @See MapObject::GetInternet().
  bool HasWifi() const { return GetInternet() == osm::Internet::Wlan; }
```
```
Internet InternetFromString(std::string_view inet)
{
  if (inet.empty())
    return Internet::Unknown;
  if (inet.find(kWlan) != string::npos)
    return Internet::Wlan;
  if (inet.find(kWired) != string::npos)
    return Internet::Wired;
  if (inet.find(kTerminal) != string::npos)
    return Internet::Terminal;
  if (inet == kYes)
    return Internet::Yes;
  if (inet == kNo)
    return Internet::No;
  return Internet::Unknown;
}
```

For toilets it doesn't make sense probably (well we can add a presence of a "changing table" later, but it could be a separate icon/line from toilets).

It might make sense for Fees if we want to display exact values (if available).

So I think the best approach is too use a subtitle icon only, unless there is an extra info that warrants an addition of a separate line.

**Comment by dvdmrtnz** [comment]
> > > What was your original intention of adding this line given that it "duplicates" the emoji? If you agree its a duplication then yes, remove it please.
> > 
> > 
> > For consistency, For example WiFi is also duplicated in description and attribute line:
> 
> Yeap, a good point. I think it was intended to display various "internet options", like wired, terminal, etc. But its stuck with a "WiFi" option only.
> 
> ```
>   /// TODO: Support all possible Internet types in UI. @See MapObject::GetInternet().
>   bool HasWifi() const { return GetInternet() == osm::Internet::Wlan; }
> ```
> 
> ```
> Internet InternetFromString(std::string_view inet)
> {
>   if (inet.empty())
>     return Internet::Unknown;
>   if (inet.find(kWlan) != string::npos)
>     return Internet::Wlan;
>   if (inet.find(kWired) != string::npos)
>     return Internet::Wired;
>   if (inet.find(kTerminal) != string::npos)
>     return Internet::Terminal;
>   if (inet == kYes)
>     return Internet::Yes;
>   if (inet == kNo)
>     return Internet::No;
>   return Internet::Unknown;
> }
> ```
> 
> For toilets it doesn't make sense probably (well we can add a presence of a "changing table" later, but it could be a separate icon/line from toilets).
> 
> It might make sense for Fees if we want to display exact values (if available).
> 
> So I think the best approach is too use a subtitle icon only, unless there is an extra info that warrants an addition of a separate line.

I agree that it is not very useful right now as it only duplicates the info.

* Maybe female/male could be added in the future? https://github.com/organicmaps/organicmaps/issues/5693

**Comment by dvdmrtnz** [comment]
> > I am going to change the approach here. Instead of lowering the priority of `amenity=toilets`, I am going to add a new type `toilets=yes` and remove them from `replaced_tags.txt`. This way we make sure that the subtag `toilets=yes` never overrides the main `amenity` tag.
> 
> I have added `toilets-yes` to `categories.txt` but the search for this type is not working. Does this need to be added somewhere else for search to work? Can I get some help here?

Nevermind. I figured it out. It needs to be added to `TypeAlwaysExists`, not to `IsUsefulNondrawableType`

**Comment by dvdmrtnz** [comment]
@pastk I will keep it simple. I have removed the attribute line in placepage. It can be added later when more toilet details are added.

File: data/categories.txt [hunk 2]
```diff
@@ -6491,6 +6491,30 @@ el:μπάνιο|χώρος ανάγκης|αποχωρητήριο|wc
 sk:3Toalety|wc
 mr:शौचालय
 
+toilets-yes|@toilet
```
Comment (vng) [inline data/categories.txt:6494]: No need a separate entry. Fix an existing one:
amenity-toilets|toilets-yes|@toilet

File: search/intermediate_result.cpp [hunk 3]
```diff
@@ -314,6 +316,8 @@ void FillDetails(FeatureType & ft, Result::Details & details)
     description.append(osm::MapObject::kFieldsSeparator).append(brand);
   if (!cuisine.empty())
     description.append(osm::MapObject::kFieldsSeparator).append(cuisine);
+  if (!toilets.empty())
+    description.append(osm::MapObject::kFieldsSeparator).append(toilets);
```
Comment (vng) [inline search/intermediate_result.cpp:320]: nit:
```
if (feature::HasToilets(typesHolder))
  description.append(osm::MapObject::kFieldsSeparator).append(osm::MapObject::kToiletsSymbol);
```

**Comment by vng** [comment]
Merge test fix commit into main "Add toilet-yes type"

File: search/intermediate_result.cpp (hunk 3 above)
Comment (dvdmrtnz) [reply search/intermediate_result.cpp:320]: Done

File: data/categories.txt (hunk 2 above)
Comment (dvdmrtnz) [reply data/categories.txt:6494]: Thanks! I didn't want to duplicate but I didn't know how to do it. I knew in `type_strings.txt` you could use the `ref =` but I didn't know how to do it in `categories.txt`

**Comment by dvdmrtnz** [comment]
> * Merge test fix commit into main "Add toilet-yes type"

Merged. 

> * Check that toilet-yes category is searchable.

It works! 👌

**Review by pastk — APPROVED** [review]

**Pull request merged**